vm/src/pyobjectrc.rs

```diff
@@ -0,0 +1,136 @@
+use std::borrow;
+use std::convert::From;
+use std::fmt;
+use std::ops::Deref;
+
+use crate::common::rc::{PyRc, PyWeak};
+use crate::pyobject::{DynPyObject, IdProtocol, PyObject, PyObjectPayload, PyRef, PyValue};
+use crate::vm::VirtualMachine;
+
+#[derive(Clone)]
+pub struct PyObjectRc {
+    inner: PyRc<DynPyObject>,
+}
+
+pub type PyObjectWeak = PyWeak<DynPyObject>;
+
+type InnerRc = PyRc<DynPyObject>;
+
+impl PyObjectRc {
+    /// # Safety
+    /// if rc is dropped without wrapping again, drop will not be called
+    unsafe fn into_rc(this: Self) -> PyRc<DynPyObject> {
+        std::mem::transmute(this)
+    }
+
+    pub fn into_raw(this: Self) -> *const DynPyObject {
+        PyRc::into_raw(unsafe { Self::into_rc(this) })
```

Comment:
I think this can be done safely with: 
```rust
let ptr = PyRc::as_ptr(&this.inner);
mem::forget(this);
ptr
```
That how Rc and Arc implements this function.